platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/internal/deterministic/DeterministicTaskSchedulerBuilder.java

```diff
@@ -66,6 +67,10 @@ public DeterministicTaskSchedulerBuilder(
     @NonNull
     @Override
     public TaskScheduler<OUT> build() {
+        // Check to ensure unhandled task capacity is not set for direct schedulers
+        if ((type == DIRECT || type == DIRECT_THREADSAFE) && unhandledTaskCapacity != 1) {
```

Comment:
this check is repeated, it makes sense to put it in a static method

Reply:
Fixed it.